### DISCUSSION TITLE
Run multiple models at the same time.

### DISCUSSION THREAD

**Discussion by Jahidur1414** [ultralytics/ultralytics#2617] [General]
I have a problem where I need to run multiple object detection models(yolov8) on a single video and the output shall also be in a single video . Performance should be close to real-time. The database can not be merged. model running should be simultaneous. How can I solve this problem?

**Comment by glenn-jocher** [comment]
@Jahidur1414 hello! Running multiple YOLOv8 models simultaneously on a single video stream and combining their outputs into one video is a challenging task, especially if you're aiming for near real-time performance. Here's a high-level approach to tackle this:

1. **Multi-Threading/Processing**: Utilize multi-threading or multi-processing in Python to run each model in parallel. This will allow you to process frames concurrently, leveraging multiple CPU cores or GPU resources if available.

2. **Synchronization**: Ensure that the frames being processed by different models are synchronized so that the outputs correspond to the same frame in the video.

3. **Combining Outputs**: After processing, you'll need to combine the detections from each model. This could involve merging bounding boxes and annotations into a single frame.

4. **Optimization**: To maintain performance, you may need to optimize your models and code. This could include using batch processing, model quantization, or hardware acceleration.

5. **Video Writing**: Finally, write the combined output frames to a new video file, ensuring that the frame rate is consistent with the input video for smooth playback.

Remember, achieving real-time performance depends on the complexity of the models, the resolution of the video, and the hardware capabilities. You might need to experiment with different configurations to find the best balance between speed and accuracy.

Good luck with your implementation! 🚀

**Comment by ashutoshbk** [comment]
Hey @Jahidur1414 , did you implement the solution?. I would love to know.

**Reply by poudelef** [reply]
I looking for solution too. Did anyone implement it?
